packages/cosmwasm-stargate/src/aminotypes.ts

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import {
+  MsgClearAdmin,
+  MsgExecuteContract,
+  MsgInstantiateContract,
+  MsgMigrateContract,
+  MsgStoreCode,
+  MsgUpdateAdmin,
+} from "@cosmjs/cosmwasm-launchpad";
+import { fromAscii, fromBase64, toAscii, toBase64 } from "@cosmjs/encoding";
+import { Coin } from "@cosmjs/launchpad";
+import { AminoConverter, codec } from "@cosmjs/stargate";
+import { assert } from "@cosmjs/utils";
+import Long from "long";
+
+import { cosmwasm } from "./codec";
+
+type ICoin = codec.cosmos.base.v1beta1.ICoin;
+type IMsgStoreCode = cosmwasm.wasm.v1beta1.IMsgStoreCode;
+type IMsgInstantiateContract = cosmwasm.wasm.v1beta1.IMsgInstantiateContract;
+type IMsgUpdateAdmin = cosmwasm.wasm.v1beta1.IMsgUpdateAdmin;
+type IMsgClearAdmin = cosmwasm.wasm.v1beta1.IMsgClearAdmin;
+type IMsgExecuteContract = cosmwasm.wasm.v1beta1.IMsgExecuteContract;
+type IMsgMigrateContract = cosmwasm.wasm.v1beta1.IMsgMigrateContract;
+
+function checkAmount(amount: readonly ICoin[] | undefined | null): readonly Coin[] {
+  assert(amount, "missing amount");
+  return amount.map((a) => {
+    assert(a.amount, "missing amount");
+    assert(a.denom, "missing denom");
+    return {
+      amount: a.amount,
+      denom: a.denom,
+    };
+  });
+}
+
+export const defaultTypes: Record<string, AminoConverter> = {
+  "/cosmwasm.wasm.v1beta1.MsgStoreCode": {
+    aminoType: "wasm/MsgStoreCode",
+    toAmino: ({ sender, wasmByteCode, source, builder }: IMsgStoreCode): MsgStoreCode["value"] => {
+      assert(sender, "missing sender");
+      assert(wasmByteCode, "missing wasmByteCode");
+      assert(typeof source === "string", "missing source");
+      assert(typeof builder === "string", "missing builder");
+      return {
+        sender: sender,
+        wasm_byte_code: toBase64(wasmByteCode),
+        source: source,
+        builder: builder,
+      };
+    },
+    fromAmino: ({ sender, wasm_byte_code, source, builder }: MsgStoreCode["value"]): IMsgStoreCode => ({
+      sender: sender,
+      wasmByteCode: fromBase64(wasm_byte_code),
+      source: source,
+      builder: builder,
+    }),
+  },
+  "/cosmwasm.wasm.v1beta1.MsgInstantiateContract": {
+    aminoType: "wasm/MsgInstantiateContract",
+    toAmino: ({
+      sender,
+      codeId,
+      label,
+      initMsg,
+      initFunds,
+      admin,
+    }: IMsgInstantiateContract): MsgInstantiateContract["value"] => {
+      assert(sender, "missing sender");
+      assert(codeId, "missing codeId");
+      assert(label, "missing label");
+      assert(initMsg, "missing initMsg");
+      return {
+        sender: sender,
+        code_id: codeId.toString(),
+        label: label,
+        init_msg: JSON.parse(fromAscii(initMsg)),
```

Comment:
`fromAscii` is fine for tests, but should be `fromUtf8` in the implementation. The JSON can contain non-ASCII characters.